java/src/jmri/jmris/AbstractThrottleServer.java

```diff
@@ -66,14 +66,8 @@ public void setThrottleFunctions(LocoAddress l, ArrayList<Boolean> fList) {
         throttleList.forEach(t -> {
             if (t.getLocoAddress() == l) {
                 for (int i = 0; i < fList.size(); i++) {
-                    try {
-                        java.lang.reflect.Method setter = t.getClass()
-                                .getMethod("setF" + i, boolean.class);
-                        setter.invoke(t, fList.get(i));
-                    } catch (java.lang.NoSuchMethodException
-                            | java.lang.IllegalAccessException
-                            | java.lang.reflect.InvocationTargetException ex1) {
-                        log.error("", ex1);
+                    t.setFunction(i, fList.get(i));
+                    if ( i > t.getFunctions().length-1) {
```

Comment:
This sets the function value and then ALWAYS sends an error status.  That isn't right.

The If statement needs to be before the `t.setFunction()` call with the send of error status as the else.

Reply:
It won't alway send the error status, only if i is greater than the number of functions on the Throttle, console error logging provided by the Throttle.setFunction() method.

I'll update to if / else though to make easier to read, it's also flagged as duplicate code so I'll move the loop to a new method.

Reply:
I think you misunderstand what I meant there.

it needs to read (in psudocode)

if(i<number of functions)
    setFunction i
else
    send error.

This is what the old code effectively did using the try/catch blocks.